icu4c/source/common/unicode/char16ptr.h

```diff
@@ -315,7 +315,7 @@ inline OldUChar *toOldUCharPtr(char16_t *p) {
 template<typename T>
 constexpr bool ConvertibleToU16StringView =
     std::is_convertible_v<T, std::u16string_view>
-#if !U_CHAR16_IS_TYPEDEF
+#if !U_CHAR16_IS_TYPEDEF && (!defined(_LIBCPP_VERSION) || _LIBCPP_VERSION < 180000)
```

Comment:
Don't you want to use 190000 as the threshold?
From your explanation, libc++ 18 merely deprecated using uint16_t, and then 19 removed it.